### DISCUSSION TITLE
[gRPC API] method for extracting all spans between two timestamps

### DISCUSSION THREAD

**Discussion by vsrakesh** [jaegertracing/jaeger#3094] [Q&A]
hi all! - I have started implementing Jeager integration into our stack recently and are also exploring the API's to fetch data using the gRPC and the HTTP APIs'

Now, I was originally intending to get all spans between two timestamp values and could not find an an direct API to accomplish this task. 
I had tried using the `findTraces` api and could get the data for each `service` separately, however this would mean that the SPAN data would be duplicated across services (as each `findTrace` retrieves spans within the service and all other related spans for each traceId associated with the service).

Would be interesting to know if Im missing something and if there are other ways of fetching raw span data (bw timestamps) via the gRPC API. 

Im trying to avoid fetching this data directly from elasticsearch

Regards

**Comment by albertteoh** [comment]
@vsrakesh what is your use case? i.e. what are you trying to accomplish? 

Why would you want to avoid directly fetching this data from Elasticsearch?

**Reply by vsrakesh** [reply]
The use case is to fetch and analyse trace data between two timestamps and to look for root cause for an issue and thresholds programatically. I'm able to do this using findTraces (and then deduplicating spans) - but would be nice to have an efficient method.
Trying to avoiding accessing the data store directly as the store could be Cassandra or Elastic and they might not be readily accessible to us.

**Reply by albertteoh** [reply]
I think [this check for a mandatory service parameter](https://github.com/jaegertracing/jaeger/blob/master/cmd/query/app/query_parser.go#L174-L176) is what's preventing you from using jaeger-query's `/traces` API to achieve what you need.

Given there is a [trace count limit in place](https://github.com/jaegertracing/jaeger/blob/master/cmd/query/app/query_parser.go#L133), I'm not sure why the service name is a required parameter.

Anyone know?
